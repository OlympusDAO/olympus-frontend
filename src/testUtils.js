import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { light as lightTheme } from "./themes/light.js";
import { Web3ContextProvider } from "./hooks/web3Context";
import store from "./store.ts";
import { createMemoryHistory } from "history";
import App from "./App";
import { I18nProvider } from "@lingui/react";
import { i18n } from "@lingui/core";

const AllTheProviders = ({ children }) => {
  return (
    <Web3ContextProvider>
      <Provider store={store}>
        <BrowserRouter basename={"/"}>
          <ThemeProvider theme={lightTheme}>
            <CssBaseline />
            {children}
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    </Web3ContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

const renderRoute = function (route) {
  const history = createMemoryHistory();
  history.push(route);
  render(
    <Web3ContextProvider>
      <Provider store={store}>
        <I18nProvider i18n={i18n}>
          <BrowserRouter basename={"/#"}>
            <App />
          </BrowserRouter>
        </I18nProvider>
      </Provider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
