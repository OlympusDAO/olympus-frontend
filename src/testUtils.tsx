import { i18n } from "@lingui/core";
import { I18nProvider } from "@lingui/react";
import CssBaseline from "@material-ui/core/CssBaseline";
import { ThemeProvider } from "@material-ui/core/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React, { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import App from "./App";
import { Web3ContextProvider } from "./hooks/web3Context";
import { ReactQueryProvider } from "./lib/react-query";
import defaultStore from "./store";
import { light as lightTheme } from "./themes/light.js";

const customRender = (ui: ReactElement, store = defaultStore, options?: RenderOptions): RenderResult => {
  const ProviderWrapper = ({ children }: { children?: ReactNode }) => (
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <ThemeProvider theme={lightTheme}>
                <CssBaseline />
                {children}
              </ThemeProvider>
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>
  );
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const renderRoute = function (route: string, store = defaultStore) {
  const history = createMemoryHistory();
  history.push(route);
  return render(
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <App />
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
