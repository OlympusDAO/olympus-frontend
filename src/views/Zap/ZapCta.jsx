import { Box, Button, Paper, Typography, Grid, SvgIcon, Link } from "@material-ui/core";
import "./zap.scss";
import { ReactComponent as ArrowUp } from "../../assets/icons/arrow-up.svg";
import HeaderLogo from "./HeaderLogo";
import { NavLink } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(theme => ({
  subHeader: {
    [theme.breakpoints.down("sm")]: {
      display: "none",
    },
    [theme.breakpoints.up("sm")]: {
      display: "flex",
    },
  },
  buttonBox: {
    [theme.breakpoints.down("sm")]: {
      justifyContent: "center",
    },
    [theme.breakpoints.up("sm")]: {
      justifyContent: "flex-end",
    },
  },
}));

function ZapCta({ tokens }) {
  const classes = useStyles();
  return (
    <>
      <Paper className="ohm-card" id="olyzaps-cta">
        <Grid container display="flex" className="cta-box" wrap>
          <Grid item xs={5} sm={3} className="icons-box">
            <HeaderLogo images={tokens} />
          </Grid>
          <Grid item xs={7} sm={5}>
            <Box alignItems="center" display="flex" flexDirection="column">
              <Typography color="textPrimary" align="center" className="cta-header">
                Zap with more assets and stake <strong>OHM</strong>
              </Typography>
              <Typography
                color="textPrimary"
                align="center"
                className="cta-subheader"
                classes={{ root: classes.subHeader }}
              >
                All-in-One contract to save time
              </Typography>
            </Box>
          </Grid>
          <Grid item xs={12} sm={4} className="button-box" classes={{ root: classes.buttonBox }}>
            <Link component={NavLink} to="/zap">
              <Button component="div" variant="outlined" color="secondary" className="learn-more-button">
                <Typography variant="body1">Swap into sOHM</Typography>
                <Box width={4} />
                <SvgIcon component={ArrowUp} color="primary" />
              </Button>
            </Link>
          </Grid>
        </Grid>
      </Paper>
    </>
  );
}

export default ZapCta;
