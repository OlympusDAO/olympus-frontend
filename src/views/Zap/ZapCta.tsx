import "./Zap.scss";

import { Trans } from "@lingui/macro";
import { Box, Button, Grid, Link, Paper, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import { TokenStack } from "@olympusdao/component-library";
import React from "react";
import { NavLink } from "react-router-dom";

const useStyles = makeStyles(theme => ({
  subHeader: {
    [theme.breakpoints.down("sm")]: {
      display: "none",
    },
    [theme.breakpoints.up("sm")]: {
      display: "flex",
    },
  },
  buttonBox: {
    [theme.breakpoints.down("sm")]: {
      justifyContent: "center",
    },
    [theme.breakpoints.up("sm")]: {
      justifyContent: "flex-end",
    },
  },
}));

const ZapCta: React.FC = () => {
  const classes = useStyles();
  return (
    <>
      <Paper className="ohm-card" id="olyzaps-cta">
        <Grid container className="cta-box">
          <Grid item xs={5} sm={3} className="icons-box">
            <TokenStack tokens={["DAI", "wETH"]} />
          </Grid>
          <Grid item xs={7} sm={5}>
            <Box alignItems="center" display="flex" flexDirection="column">
              <Typography color="textPrimary" align="center" className="cta-header">
                <Trans>
                  Zap with more assets and stake <strong>OHM</strong>
                </Trans>
              </Typography>
              <Typography
                color="textPrimary"
                align="center"
                className="cta-subheader"
                classes={{ root: classes.subHeader }}
              >
                <Trans>All-in-One contract to save time</Trans>
              </Typography>
            </Box>
          </Grid>
          <Grid item xs={12} sm={4} className="button-box" classes={{ root: classes.buttonBox }}>
            <Link component={NavLink} to="/zap">
              <Button component="div" variant="outlined" color="secondary" className="learn-more-button">
                <Typography variant="body1">
                  <Trans>Swap into sOHM</Trans>
                </Typography>
              </Button>
            </Link>
          </Grid>
        </Grid>
      </Paper>
    </>
  );
};

export default ZapCta;
