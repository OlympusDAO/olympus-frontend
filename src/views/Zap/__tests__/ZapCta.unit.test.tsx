import { render } from "../../../testUtils";
import ZapCta from "../ZapCta";

describe("<ZapCta/>", () => {
  it("should render component", () => {
    const { container } = render(<ZapCta />);
    expect(container).toMatchSnapshot();
  });
});
