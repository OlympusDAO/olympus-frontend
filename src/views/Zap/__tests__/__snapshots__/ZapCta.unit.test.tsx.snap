// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ZapCta/> should render component 1`] = `
<div>
  <div
    class="MuiPaper-root ohm-card MuiPaper-elevation0 MuiPaper-rounded"
    id="olyzaps-cta"
  >
    <div
      class="MuiGrid-root cta-box MuiGrid-container"
    >
      <div
        class="MuiGrid-root icons-box MuiGrid-item MuiGrid-grid-xs-5 MuiGrid-grid-sm-3"
      >
        <div
          class="MuiBox-root MuiBox-root-3"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            style="z-index: 1;"
            viewBox="0 0 32 32"
          >
            <circle
              cx="16"
              cy="16"
              fill="#F5AC37"
              r="16"
            />
            <path
              d="M16.59 17.13h6.08c.13 0 .19 0 .2-.17.05-.619.05-1.24 0-1.86 0-.12-.06-.17-.19-.17h-12.1c-.15 0-.19.05-.19.19v1.78c0 .23 0 .23.24.23h5.96Zm5.6-4.28a.195.195 0 0 0 0-.14 3.96 3.96 0 0 0-.36-.63 5.103 5.103 0 0 0-.74-.93 2.451 2.451 0 0 0-.461-.45 7.187 7.187 0 0 0-3-1.52 7.458 7.458 0 0 0-1.7-.18h-5.37c-.15 0-.17.06-.17.19v3.55c0 .15 0 .19.19.19h11.54s.1-.02.12-.08h-.049Zm0 6.36a2.336 2.336 0 0 0-.51 0H10.59c-.15 0-.2 0-.2.2v3.47c0 .16 0 .2.2.2h5.12c.245.02.49.002.73-.049a7.626 7.626 0 0 0 2.17-.48c.253-.088.498-.203.73-.34h.07a6.496 6.496 0 0 0 2.79-2.809s.07-.151-.01-.19v-.002ZM8.38 24.88v-5.53c0-.13 0-.15-.16-.15H6.05c-.12 0-.17 0-.17-.16v-1.9H8.2c.13 0 .18 0 .18-.17v-1.88c0-.12 0-.149-.16-.149H6.05c-.12 0-.17 0-.17-.16v-1.76c0-.11 0-.14.16-.14h2.15c.15 0 .19 0 .19-.19v-5.39c0-.16 0-.2.2-.2h7.5c.544.022 1.085.082 1.62.18a9.775 9.775 0 0 1 3.13 1.16 8.82 8.82 0 0 1 1.76 1.36c.396.412.753.857 1.07 1.33.314.48.575.994.78 1.53a.26.26 0 0 0 .3.21h1.79c.23 0 .23 0 .24.221v1.64c0 .16-.06.2-.221.2h-1.38c-.14 0-.18 0-.17.18.054.61.054 1.22 0 1.83 0 .17 0 .19.19.19h1.58c.07.09 0 .18 0 .27.01.117.01.234 0 .35v1.21c0 .17-.05.22-.2.22h-1.89a.25.25 0 0 0-.29.19 7.997 7.997 0 0 1-2.101 3.06c-.34.307-.697.595-1.07.86-.4.23-.79.47-1.2.66-.756.34-1.548.591-2.36.75a12.3 12.3 0 0 1-2.34.19h-6.94v-.01l.002-.002Z"
              fill="#FEFEFD"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            style="margin-left: -6px; z-index: 1;"
            viewBox="0 0 32 32"
          >
            <circle
              cx="16"
              cy="16"
              fill="#fff"
              r="15"
              stroke="url(#wETH_svg__a)"
              stroke-width="2"
            />
            <path
              clip-rule="evenodd"
              d="M16.25 20.976 10 17.349 16.25 26V26l6.253-8.65-6.253 3.626Z"
              fill="#708B96"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="m16.25 6 6.248 10.186-6.248-2.793L10 16.186 16.25 6Zm0 7.395L10 16.185l6.25 3.629 6.248-3.628-6.248-2.791Z"
              fill="#424242"
              fill-rule="evenodd"
            />
            <defs>
              <lineargradient
                gradientUnits="userSpaceOnUse"
                id="wETH_svg__a"
                x1="16"
                x2="16"
                y1="0"
                y2="32"
              >
                <stop
                  stop-color="#444243"
                />
                <stop
                  offset="1"
                  stop-color="#708B96"
                />
              </lineargradient>
            </defs>
          </svg>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 MuiGrid-grid-sm-5"
      >
        <div
          class="MuiBox-root MuiBox-root-5"
        >
          <p
            class="MuiTypography-root cta-header MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
          >
            Zap with more assets and stake 
            <strong>
              OHM
            </strong>
          </p>
          <p
            class="MuiTypography-root makeStyles-subHeader-1 cta-subheader MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
          >
            All-in-One contract to save time
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-buttonBox-2 button-box MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
          href="/zap"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined learn-more-button MuiButton-outlinedSecondary MuiButton-disableElevation"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Swap into sOHM
              </p>
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
